src/qat/purr/compiler/runtime.py

```diff
@@ -193,7 +193,7 @@ def _common_execute(
         if self.engine is None:
             raise ValueError("No execution engine available.")
 
-        if isinstance(instructions, InstructionBuilder):
+        if isinstance(instructions, InstructionBuilder) or any([i for i in instructions if isinstance(i, InstructionBuilder)]):
```

Comment:
directly `... or isinstance(instructions, FluidInstructionBuilder)`, otherwise you're not guaranteed it'll have a `instructions` field

Reply:
Ben just mentioned that to me as well! Fixed